qt/python/mantidqt/widgets/codeeditor/completion.py

```diff
@@ -0,0 +1,185 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+#  This file is part of the mantid workbench.
+
```

Comment:
I think it might be worth a module docstring with a brief comment as to why we have our own implementation and are not using something like `jedi`. Someone is bound to ask and it would be good to point them at this.